File: api/model/syllabus.go
package model

type SyllabusStructure struct {
	ID     int    `json:"id"`
	PrevID *int   `json:"prevId" db:"prev_id"`
	Title  string `json:"title"`
}

type BaseSyllabus struct {
	ID    int    `json:"id"`
	Title string `json:"title"`
}

type Syllabus struct {
	BaseSyllabus
	ParentID    *int `json:"parentId" db:"parent_id"`
	StructureID *int `json:"structureId,omitempty" db:"structure_id"`
}

File: web/src/types/syllabus.ts
import type { User } from './user';

export type Structure = {
  id: number;
  prevId: number | null;
  title: string;
};

export type BaseSyllabus = {
  id: number;
  title: string;
};

export type Syllabus = BaseSyllabus & {
  parentId: number | null;
  structureId?: number;
};

export type Score = {
  user: User;
  score: number | null;
};
